src/basic/os-util.c

```diff
@@ -225,16 +247,16 @@ int open_extension_release(const char *root, const char *extension, bool relax_e
         return 0;
 }
 
-int fopen_extension_release(const char *root, const char *extension, bool relax_extension_release_check, char **ret_path, FILE **ret_file) {
+int fopen_extension_release(const char *root, ImageClass image_class, const char *extension, bool relax_extension_release_check, char **ret_path, FILE **ret_file) {
         _cleanup_free_ char *p = NULL;
         _cleanup_close_ int fd = -EBADF;
         FILE *f;
         int r;
 
```

Comment:
all these calls should check that image_class is actually valid for extensions. i.e. maybe add `assert(image_class_release_info[image_class].release_file_name_prefix);` here.

also, add this first:

```c
assert(image_class >= 0);
assert(image_class < _IMAGE_CLASS_MAX);
```

Reply:
this is not resolved